src/components/VmActions/index.js

```diff
@@ -96,6 +87,65 @@ VmDropdownActions.propTypes = {
   id: PropTypes.string.isRequired,
 }
 
+export function getConsoleActions ({ vm, msg, onOpenConsole, idPrefix, config, preferredConsole }) {
+  const vncConsole = vm.get('consoles').find(c => c.get('protocol') === VNC)
+  const spiceConsole = vm.get('consoles').find(c => c.get('protocol') === SPICE)
+  const hasRdp = isWindows(vm.getIn(['os', 'type']))
+  let consoles = []
+
+  if (vncConsole) {
+    const vncModes = [{
+      priority: 0,
+      protocol: VNC,
+      consoleType: NATIVE_VNC,
+      shortTitle: msg.vncConsole(),
+      icon: <Icon name='external-link' />,
+      id: `${idPrefix}-button-console-vnc`,
+      onClick: () => { onOpenConsole({ consoleType: NATIVE_VNC }) },
+    },
+    {
+      priority: 0,
+      consoleType: BROWSER_VNC,
+      shortTitle: msg.vncConsoleBrowser(),
+      actionDisabled: config.get('websocket') === null,
+      id: `${idPrefix}-button-console-browser`,
+      onClick: () => { onOpenConsole({ consoleType: BROWSER_VNC }) },
+    }]
+
+    if (config.get('defaultVncMode') === NO_VNC) {
+      vncModes.reverse()
+    }
+    consoles = [...consoles, ...vncModes]
```

Comment:
Don't need to use `...` everywhere

```suggestion
    if (config.get('defaultVncMode') === NO_VNC) {
      consoles.push(browserVnc, nativeVnc)
    } else {
      consoles.push(nativeVnc, browserVnc)
    }
```